common/utils/src/main/scala/org/apache/spark/util/IvyTestUtils.scala

```diff
@@ -330,58 +341,78 @@ private[deploy] object IvyTestUtils {
+   *   whether to mock the Ivy layout for local repository testing
+   * @param withPython
+   *   Whether to pack python files inside the jar for extensive testing.
+   * @return
+   *   Root path of the repository. Will be `rootDir` if supplied.
    */
-  private[deploy] def createLocalRepositoryForTests(
+  private[spark] def createLocalRepositoryForTests(
       artifact: MavenCoordinate,
       dependencies: Option[String],
       rootDir: Option[File],
       useIvyLayout: Boolean = false,
       withPython: Boolean = false,
       withR: Boolean = false): File = {
-    val deps = dependencies.map(SparkSubmitUtils.extractMavenCoordinates)
+    val deps = dependencies.map(MavenUtils.extractMavenCoordinates)
     val mainRepo = createLocalRepository(artifact, deps, rootDir, useIvyLayout, withPython, withR)
-    deps.foreach { seq => seq.foreach { dep =>
-      createLocalRepository(dep, None, Some(mainRepo), useIvyLayout, withPython = false)
-    }}
+    deps.foreach { seq =>
+      seq.foreach { dep =>
+        createLocalRepository(dep, None, Some(mainRepo), useIvyLayout, withPython = false)
+      }
+    }
     mainRepo
   }
 
   /**
    * Creates a repository for a test, and cleans it up afterwards.
    *
-   * @param artifact The main maven coordinate to generate the jar and pom for.
-   * @param dependencies List of dependencies this artifact might have to also create jars and poms.
-   * @param rootDir The root folder of the repository (like `~/.m2/repositories`)
-   * @param useIvyLayout whether to mock the Ivy layout for local repository testing
-   * @param withPython Whether to pack python files inside the jar for extensive testing.
-   * @return Root path of the repository. Will be `rootDir` if supplied.
+   * @param artifact
+   *   The main maven coordinate to generate the jar and pom for.
+   * @param dependencies
+   *   List of dependencies this artifact might have to also create jars and poms.
+   * @param rootDir
+   *   The root folder of the repository (like `~/.m2/repositories`)
+   * @param useIvyLayout
+   *   whether to mock the Ivy layout for local repository testing
+   * @param withPython
+   *   Whether to pack python files inside the jar for extensive testing.
+   * @return
+   *   Root path of the repository. Will be `rootDir` if supplied.
    */
-  private[deploy] def withRepository(
+  private[spark] def withRepository(
       artifact: MavenCoordinate,
       dependencies: Option[String],
       rootDir: Option[File],
       useIvyLayout: Boolean = false,
       withPython: Boolean = false,
       withR: Boolean = false,
       ivySettings: IvySettings = new IvySettings)(f: String => Unit): Unit = {
-    val deps = dependencies.map(SparkSubmitUtils.extractMavenCoordinates)
+    val deps = dependencies.map(MavenUtils.extractMavenCoordinates)
     purgeLocalIvyCache(artifact, deps, ivySettings)
-    val repo = createLocalRepositoryForTests(artifact, dependencies, rootDir, useIvyLayout,
```

Comment:
Did you run scalafmt against this whole file? Let's avoid doing that. This Ivy has a bunch of legacy code, and should be best to avoid unrelated changes here.

Reply:
Sure, reverted scalafmt changes not related to the actual PR changes